packages/react/src/components/table/Table.test.tsx

```diff
@@ -102,6 +102,43 @@ describe('<Table /> spec', () => {
     expect(ageOfSortedTableFirstRow).toHaveTextContent('8');
   });
 
+  it('Should successfully call onSort when sort button is pressed', () => {
+    const colsWithSorting = [
+      { key: 'id', headerName: 'Not rendered' },
+      { key: 'firstName', headerName: 'First name', isSortable: true },
+      { key: 'surname', headerName: 'Surname', isSortable: true },
+      {
+        key: 'age',
+        headerName: 'Age',
+        transform: ({ age }) => {
+          return <div style={{ textAlign: 'right' }}>{age}</div>;
+        },
+        isSortable: true,
+      },
+      { key: 'profession', headerName: 'Profession', isSortable: true },
+    ];
+
+    const mockOnSort = jest.fn();
+
+    const { container } = render(
+      <Table
+        ariaLabelSortButtonUnset="Not sorted"
+        ariaLabelSortButtonAscending="Sorted in ascending order"
+        ariaLabelSortButtonDescending="Sorted in descending order"
+        caption={caption}
+        cols={colsWithSorting}
+        rows={rows}
+        indexKey={indexKey}
+        onSort={mockOnSort}
+        renderIndexCol={renderIndexCol}
+      />,
+    );
+
+    userEvent.click(container.querySelector('[data-testid="hds-table-sorting-header-age"]'));
+
+    expect(mockOnSort).toHaveBeenCalledTimes(1);
```

Comment:
This could be tested without mock by:
1. Providing initial rows, for example, names could be D, E, F
2. Providing a custom onSort which replaces those initial rows with rows from "server" that have names A, B, C
3. User clicks name sort
4. Asserting that the server rows are visible or have replaced the initial rows

Reply:
I really like this as we talked just this morning about avoiding mocking in tests but should we make this test in a separate branch after the release? Maybe look for other mocks and refactor them as well.